lib/src/parser.h

```diff
@@ -31,6 +31,11 @@ typedef struct {
   uint16_t length;
 } TSFieldMapSlice;
 
+typedef struct {
+  uint16_t index;
+  uint16_t length;
+} TSSupertypeMapSlice;
```

Comment:
Instead of duplicating a type definition, rename `TSFieldMapSlice` to `TSMapSlice`, and leave a comment explaining that it's used for looking up the index and length of either the field map slices or supertype map slices.

Reply:
Will the renaming of the `TSFieldMapSlice` be breaking when generating for a previous ABI?

Reply:
Nah it shouldn't, because the layout didn't change.

If you're compiling a parser generated with the type renamed in parser.c, and you're using an old header, obviously that'd break, but that *shouldn't* happen, because everytime you regenerate, everything in `src` is "regenerated". So, they should never be "out of sync", and you can use a parser compiled with the old name and with the new name in the same application, because the layout is the exact same.

Reply:
Ok cool thanks, that makes sense